Python/Product/TestAdapter/TestContainerDiscovererWorkspace.cs

```diff
@@ -177,8 +177,7 @@ class TestContainerDiscovererWorkspace : ITestContainerDiscoverer, IDisposable {
                 _testFilesUpdateWatcher.AddDirectoryWatch(workspace.Location);
                 oldWatcher?.Dispose();
 
-                Regex testFileFilterRegex = new Regex(@".*\.(py|txt)", RegexOptions.Compiled | RegexOptions.IgnoreCase);
-                Predicate<string> testFileFilter = (x) => testFileFilterRegex.IsMatch(x);
+                Predicate<string> testFileFilter = (x) => PythonConstants.TestFileExtensionRegex.IsMatch(x);
```

Comment:
do you need to call `IsMatch(PathUtils.GetFileOrDirectoryName(x))` here?

Reply:
It's optional. The regex will only look at the last few characters of the string to check if it matches ".txt" or ".py", so it doesn't matter if the string contains the full path or just the file name. I've included it in the new commit so it's easier to read and understand.